charts/ingress-nginx/templates/controller-service-internal.yaml

```diff
@@ -23,10 +29,16 @@ spec:
 {{- end }}
 {{- if .Values.controller.service.internal.externalTrafficPolicy }}
   externalTrafficPolicy: {{ .Values.controller.service.internal.externalTrafficPolicy }}
+{{- end }}
+{{- if .Values.controller.service.internal.sessionAffinity }}
+  sessionAffinity: {{ .Values.controller.service.internal.sessionAffinity }}
+{{- end }}
+{{- if .Values.controller.service.internal.healthCheckNodePort }}
+  healthCheckNodePort: {{ .Values.controller.service.internal.healthCheckNodePort }}
 {{- end }}
   ports:
   {{- $setNodePorts := (or (eq .Values.controller.service.type "NodePort") (eq .Values.controller.service.type "LoadBalancer")) }}
-  {{- if .Values.controller.service.enableHttp }}
+  {{- if or .Values.controller.service.internal.enableHttp (and (not (hasKey .Values.controller.service.internal "enableHttp")) .Values.controller.service.enableHttp) }}
```

Comment:
I am not following the goal of the if statements. Why are you checking hasKey on `service.internal.enableHttp`?

Reply:
The goal is to do something like this:
`{{- if default .Values.controller.service.enableHttps .Values.controller.service.internal.enableHttps }}`
Due to limitation with default function (https://github.com/helm/helm/issues/3308) I need to do this ugly if statement.
I'm open to any better proposal.

Reply:
Yikes, that is quite an issue. Could you add a comment before the first if statement directing to that issue. Hopefully, it will be fixed in the future so that we can make this more readable.